### PR TITLE
Add PyTorch 2.5 to regression test

### PR DESCRIPTION
**Pull request by leslie-fang-intel**
Replace https://github.com/pytorch/ao/pull/1100

### CODE REVIEW

**Comment by leslie-fang-intel** [comment]
cc @gau-nernst @msaroufim @jerryzh168

**Comment by msaroufim** [comment]
I think this is mergeable since we pinned nightly PyTorch. Mind just rebasing to mai?

**Review by gau-nernst — APPROVED** [review]

**Review by msaroufim — APPROVED** [review]

File: test/integration/test_integration.py [hunk 1]
```diff
@@ -104,7 +105,13 @@
 def _int8wo_api(mod):
     if TORCH_VERSION_AT_LEAST_2_4:
         quantize_(mod, int8_weight_only(), set_inductor_config=False)
-        if not TORCH_VERSION_AT_LEAST_2_5:
+        if (
```
Comment (jerryzh168) [inline test/integration/test_integration.py:108]: wow condition is becoming pretty complicated..

Comment (leslie-fang-intel) [reply test/integration/test_integration.py:108]: yean, ensure as much as test combination has been covered.

**Pull request merged**